pkg/controller-gen/generators/group_version_interface_go.go

```diff
@@ -93,7 +93,7 @@ func (f *groupInterfaceGo) Init(c *generator.Context, w io.Writer) error {
 		body := `
 		func (v *version) {{.type}}() {{.type}}Controller {
 			return &{{.type}}GenericController {
-				generic.New{{ if not .namespaced}}NonNamespaced{{end}}Controller[*{{.version}}.{{.type}}, *{{.version}}.{{.type}}List](schema.GroupVersionKind{Group: "{{.group}}", Version: "{{.version}}", Kind: "{{.type}}"}, "{{.pluralLower}}", {{ if .namespaced}}true, {{end}}v.controllerFactory),
+				generic.New{{ if not .namespaced}}NoNs{{end}}Controller[*{{.version}}.{{.type}}, *{{.version}}.{{.type}}List](schema.GroupVersionKind{Group: "{{.group}}", Version: "{{.version}}", Kind: "{{.type}}"}, "{{.pluralLower}}", {{ if .namespaced}}true, {{end}}v.controllerFactory),
```

Comment:
Below are the differences between the two. I changed this after I had to type it out a couple of times. Honestly, both names are very verbose because you have to specify the types. So you are probably right in that we are not saving much space.
```golang
fake.NewMockNoNsControllerInterface[*v1.Namespace, *v1.NamespaceList](ctrl)
generic.NewNoNsController[*v1.Namespace, *v1.NamespaceList](schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Namespace"}, "namespaces", v.controllerFactory)

fake.NewMockNonNamespacedControllerInterface[*v1.Namespace, *v1.NamespaceList](ctrl)
generic.NewNonNamespacedController[*v1.Namespace, *v1.NamespaceList](schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Namespace"}, "namespaces", v.controllerFactory)
```